pkg/endpointmanager/manager.go

```diff
@@ -460,6 +464,35 @@ func (mgr *EndpointManager) AddEndpoint(owner regeneration.Owner, ep *endpoint.E
 	return nil
 }
 
+func (mgr *EndpointManager) AddHostEndpoint(ctx context.Context, owner regeneration.Owner,
+	proxy endpoint.EndpointProxy, allocator cache.IdentityAllocator, reason string, nodeName string) (err error) {
+	ep := endpoint.CreateHostEndpoint(owner, proxy, allocator)
+	if err := mgr.AddEndpoint(owner, ep, reason); err != nil {
+		return err
+	}
+
+	epLabels := labels.Labels{}
+	epLabels.MergeLabels(labels.LabelHost)
+
+	if k8s.IsEnabled() {
+		// Retrieve k8s labels.
+		if k8sNode, err := k8s.GetNode(k8s.Client(), nodeName); err != nil {
+			log.WithError(err).Warning("Kubernetes node resource representing own node is not available, cannot set Labels")
+		} else {
+			newLabels := labels.Map2Labels(k8sNode.GetLabels(), labels.LabelSourceK8s)
+			newIdtyLabels, _ := labelsfilter.Filter(newLabels)
+			epLabels.MergeLabels(newIdtyLabels)
+		}
+	}
+
+	// Give the endpoint a security identity
+	newCtx, cancel := context.WithTimeout(ctx, 30*time.Second)
+	defer cancel()
+	ep.UpdateLabels(newCtx, epLabels, nil, true)
```

Comment:
Do we need to put this logic into a `pkg/controller` so that if the user updates the node labels, we will recognize and apply the changes? (Or are we listening for node events and handling that case from events somehow?)

Reply:
There is a separate commit, later in the series (`pkg/identity: Watch and update labels for the host`), that adds a k8s watcher (only for the local node) for node label updates.